pkg/sdk/secrets_gen.go

```diff
@@ -0,0 +1,181 @@
+}
+type SetForGenericString struct {
+	SecretString *string `ddl:"parameter,single_quotes" sql:"SECRET_STRING"`
+}
+type SecretUnset struct {
+	Comment *bool `ddl:"keyword" sql:"SET COMMENT = NULL"`
+}
+
+// DropSecretOptions is based on https://docs.snowflake.com/en/sql-reference/sql/drop-secret.
+type DropSecretOptions struct {
+	drop     bool                   `ddl:"static" sql:"DROP"`
+	secret   bool                   `ddl:"static" sql:"SECRET"`
+	IfExists *bool                  `ddl:"keyword" sql:"IF EXISTS"`
+	name     SchemaObjectIdentifier `ddl:"identifier"`
+}
+
+// ShowSecretOptions is based on https://docs.snowflake.com/en/sql-reference/sql/show-secret.
+type ShowSecretOptions struct {
+	show    bool  `ddl:"static" sql:"SHOW"`
+	secrets bool  `ddl:"static" sql:"SECRETS"`
+	Like    *Like `ddl:"keyword" sql:"LIKE"`
+	In      *In   `ddl:"keyword" sql:"IN"`
+}
+type secretDBRow struct {
+	CreatedOn     string         `db:"created_on"`
+	Name          string         `db:"name"`
+	SchemaName    string         `db:"schema_name"`
+	DatabaseName  string         `db:"database_name"`
+	Owner         string         `db:"owner"`
+	Comment       sql.NullString `db:"comment"`
+	SecretType    string         `db:"secret_type"`
+	OauthScopes   sql.NullString `db:"oauth_scopes"`
+	OwnerRoleType string         `db:"owner_role_type"`
+}
+type Secret struct {
+	CreatedOn     string
+	Name          string
+	SchemaName    string
+	DatabaseName  string
+	Owner         string
+	Comment       string
+	SecretType    string
+	OauthScopes   sql.NullString
+	OwnerRoleType string
+}
+
+// DescribeSecretOptions is based on https://docs.snowflake.com/en/sql-reference/sql/desc-secret.
+type DescribeSecretOptions struct {
+	describe bool                   `ddl:"static" sql:"DESCRIBE"`
+	secret   bool                   `ddl:"static" sql:"SECRET"`
+	name     SchemaObjectIdentifier `ddl:"identifier"`
+}
+type secretDetailsDBRow struct {
+	CreatedOn                   string         `db:"created_on"`
+	Name                        string         `db:"name"`
+	SchemaName                  string         `db:"schema_name"`
+	DatabaseName                string         `db:"database_name"`
+	Owner                       string         `db:"owner"`
+	Comment                     sql.NullString `db:"comment"`
+	SecretType                  string         `db:"secret_type"`
+	Username                    sql.NullString `db:"username"`
+	OauthAccessTokenExpiryTime  *time.Time     `db:"oauth_access_token_expiry_time"`
+	OauthRefreshTokenExpiryTime *time.Time     `db:"oauth_refresh_token_expiry_time"`
+	OauthScopes                 sql.NullString `db:"oauth_scopes"`
+	IntegrationName             sql.NullString `db:"integration_name"`
+}
+type SecretDetails struct {
+	CreatedOn                   string
+	Name                        string
+	SchemaName                  string
+	DatabaseName                string
+	Owner                       string
+	Comment                     sql.NullString
```

Comment:
These `sql.*` types should be present only in dbRows - we want to separate things returned from Snowflake from things returned by the SDK. You can use `*string` here.